features/admin.actions.v1/components/action-config-form.tsx

```diff
@@ -0,0 +1,829 @@
+    );
+
+    /**
+     * This is called when the Change Authentication button is pressed.
+     */
+    const handleAuthenticationChange = (): void => {
+        setIsAuthenticationUpdateFormState(true);
+    };
+
+    /**
+     * This is called when the cancel button is pressed.
+     */
+    const handleAuthenticationChangeCancel = (): void => {
+        setAuthenticationType(initialValues?.authenticationType as AuthenticationType);
+        setIsAuthenticationUpdateFormState(false);
+    };
+
+    const handleSuccess = (operation: string): void => {
+        dispatch(
+            addAlert({
+                description: t("console:manage.features.actions.notification.success." + operation + ".description"),
+                level: AlertLevels.SUCCESS,
+                message: t("console:manage.features.actions.notification.success." + operation + ".message")
+            })
+        );
+    };
+
+    const handleError = (error: AxiosError, operation: string): void => {
+        if (error.response && error.response.data && error.response.data.description) {
+            dispatch(
+                addAlert({
+                    description: t("console:manage.features.actions.notification.error." + operation + ".description",
+                        { description: error.response.data.description }),
+                    level: AlertLevels.ERROR,
+                    message: t("console:manage.features.actions.notification.error." + operation + ".message")
+                })
+            );
+        } else {
+            // Generic error message
+            dispatch(
+                addAlert({
+                    description: t("console:manage.features.actions.notification.genericError." + operation
+                        + ".description"),
+                    level: AlertLevels.ERROR,
+                    message: t("console:manage.features.actions.notification.genericError." + operation
+                        + ".message")
+                })
+            );
+        }
+    };
+
+    const getFieldDisabledStatus = (): boolean => {
+        if (isCreateFormState) {
+            return !hasActionCreatePermissions;
+        } else {
+            return !hasActionUpdatePermissions;
+        }
+    };
+
+    const validateForm = (values: ActionConfigFormPropertyInterface): ActionConfigFormPropertyInterface => {
+        const error: ActionConfigFormPropertyInterface = {
+            accessTokenAuthProperty: undefined,
+            authenticationType: undefined,
+            endpointUri: undefined,
+            headerAuthProperty: undefined,
+            id: undefined,
+            name: undefined,
+            passwordAuthProperty: undefined,
+            usernameAuthProperty: undefined,
+            valueAuthProperty: undefined
+        };
+
+        if (!values?.name) {
+            error.name = t("console:manage.features.actions.fields.name.validations.empty");
+        }
+        const vsCharRegex: RegExp = /^[A-Za-z0-9 -_]*$/;
```

Comment:
Shall we use a meaningful name here `vsCharRegex`

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/6722/commits/5e3564e0797c09f4f20b71837b928bc9fa57c847